### ISSUE TITLE
[Assigned] Adaptive Icon not rendered correctly when mask is not circle

### ISSUE THREAD

**Issue by keima** [DroidKaigi/conference-app-2018#301]
## Overview (Required)

With great contribution by @mhidaka and @chigichan24 , DroidKaigi app now with Adaptive Icon. Thanks!
By the way, White space is visible at the corners when set change icon style to rounded square mode at launcher app.
<img src="https://user-images.githubusercontent.com/867470/35130925-b55b81e4-fd06-11e7-920e-bc0019ec0dea.png" width="400" />

I suspect that background drawable protruded outside (see: https://medium.com/google-design/designing-adaptive-icons-515af294c783 )
( blue line is safe zone. Actually, safe zone is circle, not square )
![2018-01-19 11 35 12](https://user-images.githubusercontent.com/867470/35132116-023c8fac-fd0d-11e7-9a7b-efbe2164361d.png)

chigichan24 said about this is intended by mhidaka at #270.
But my proposal may breaks mhidaka's intent.
Would you forgive me about it?

**Comment by mhidaka** [comment]
Welcome to contribute! 😄

**Comment by chigichan24** [comment]
Thank you for your bug report, @keima . Just as you said, the icon should be masked by circle. 
 I want to try this issue again🙋

**Comment by konifar** [comment]
Cool 🆒  I assigned!

**Comment by chigichan24** [comment]
I have a question.

I'll resolve this problem extracting background white area. like this.

![fig](https://user-images.githubusercontent.com/7840108/35142284-370221dc-fd41-11e7-8b80-4ec3388ff6e0.png)

But I think this solution is not smart.  If posible, I want to create icon dynamically.
If mask type is circle, apply the icon as until now. If mask type in not circle, apply the icon which uses in less than API Level 25. 

Does anyone have any ideas?

**Comment by keima** [comment]
@chigichan24 
tl;dr: As Nick Butcher's article ( https://medium.com/google-design/designing-adaptive-icons-515af294c783 ), We concern about "keylines".

Okay, so your suggestion is "app is only use adaptive icon if launcher use circle mask. But app use round icon (`legacy` icon) if launcher doesn't use circle", isn't it?
Unfortunately it is not provided as far as I know. Anyway, I think it's troublesome.

I actually wrote issue title "Adaptive Icon not rendered correctly when mask is not circle", 
but I know "Impossible that always keeping circle icon whether adaptive icon or not".
So we think a method that circle designed icon and Adaptive Icon existing together.

Looking below figure (blue line: 72dp square safe zone, red circle line: 66dp circle safe zone, purple circle line: keylines 52dp):
![ic_launcher_background 2018-01-19 18-04-01](https://user-images.githubusercontent.com/867470/35143538-39f26c36-fd45-11e7-83bc-4e8f4095ec20.png)
I create image that droid circle fits keyline circle.
Hmm I think it's inside of safezone perfectly.

I suggest you that droid icon fits safezone.

P.S.
If you have more time, it's time to trying png to VectorDrawable!
I'm DroidKaigi staff so I create svg icon file for you. ;-)
Would you like trying it?

**Comment by keima** [comment]
Oh, tranparent area on screenshot is not actual.
In adaptive icon, background drawable is opaque (contains no transparent).

**Comment by keima** [comment]
@chigichan24 
Yes, that's right.

Thank you for taking on it. I changed assignee to chigichan from me.
And assets is available here: https://drive.google.com/file/d/1Jw8vZCtPZTMa5b-CJYwjcKwzbS-u3Cpg/view?usp=sharing
宜しくお願いします。

Note:
Original illustrator file (and round icon created by same illustrator file) has shadow at droid hands.
But svg can't contains vector shadow so VectorDrawable lose shadow data.
To avoid it, I attached png foreground data into zip.
I think full VectorDrawable version is difficult if shadow is important, but [some app has png foreground](https://github.com/nickbutcher/plaid/blob/master/app/src/main/res/mipmap-xhdpi-v26/ic_launcher_foreground.png). 
I respect your decision whether keep shadow ( = use png ) or not.

**Comment by chigichan24** [comment]
Thank you for the BIG help @keima !!!!!!

**Comment by keima** [comment]
Yeah! Great contribution @chigichan24 ! 👏 👏 👏
